src/keyring/KeyringController.test.ts

```diff
@@ -1094,4 +1096,264 @@ describe('KeyringController', () => {
+      );
+      expect(defaultAccount.balance).toBe('0x0');
+    });
+
+    it('should sign a message with a Ledger keyring', async () => {
+      const confirmSignatureStub = sinon.stub(ledgerKeyring, 'signMessage');
+
+      confirmSignatureStub.resolves(
+        '4cb25933c5225f9f92fc9b487451b93bc3646c6aa01b72b01065b8509ac4fd6c37798695d0d5c0949ed10c5e102800ea2b62c2b670729c5631c81b0c52002a641b',
+      );
+
+      const data =
+        '0x879a053d4800c6354e76c7985a865d2922c82fb5b3f4577b2fe08b998954f2e0';
+
+      const account = await ledgerKeyring?.getDefaultAccount();
+      const signature = await keyringController.signMessage({
+        data,
+        from: account,
+      });
+      expect(signature).toBe(
+        '4cb25933c5225f9f92fc9b487451b93bc3646c6aa01b72b01065b8509ac4fd6c37798695d0d5c0949ed10c5e102800ea2b62c2b670729c5631c81b0c52002a641b',
+      );
+
+      confirmSignatureStub.reset();
+    });
+
+    it('should sign a personal message with a Ledger keyring', async () => {
+      const confirmSignatureStub = sinon.stub(
+        ledgerKeyring,
+        'signPersonalMessage',
+      );
+
+      confirmSignatureStub.resolves(
+        '4cb25933c5225f9f92fc9b487451b93bc3646c6aa01b72b01065b8509ac4fd6c37798695d0d5c0949ed10c5e102800ea2b62c2b670729c5631c81b0c52002a641b',
+      );
+
+      const data =
+        '0x879a053d4800c6354e76c7985a865d2922c82fb5b3f4577b2fe08b998954f2e0';
+
+      const account = await ledgerKeyring?.getDefaultAccount();
+      const signature = await keyringController.signPersonalMessage({
+        data,
+        from: account,
+      });
+      expect(signature).toBe(
+        '4cb25933c5225f9f92fc9b487451b93bc3646c6aa01b72b01065b8509ac4fd6c37798695d0d5c0949ed10c5e102800ea2b62c2b670729c5631c81b0c52002a641b',
+      );
+
+      confirmSignatureStub.reset();
+    });
+
+    it('should sign transaction with Ledger keyring', async () => {
+      const tx = TransactionFactory.fromTxData(
+        {
+          accessList: [],
+          chainId: '0x4',
+          data: '0x',
+          gasLimit: '0x5208',
+          maxFeePerGas: '0x2540be400',
+          maxPriorityFeePerGas: '0x3b9aca00',
+          nonce: '0x68',
+          r: undefined,
+          s: undefined,
+          to: '0x0c54fccd2e384b4bb6f2e405bf5cbc15a017aafb',
+          v: undefined,
+          value: '0x0',
+          type: 2,
+        },
+        {
+          common: Common.forCustomChain(
+            MAINNET,
+            {
+              name: 'rinkeby',
+              chainId: parseInt('4'),
+              networkId: parseInt('4'),
+            },
+            'london',
+          ),
+        },
+      );
+
+      const confirmSignatureStub = sinon.stub(ledgerKeyring, 'signTransaction');
+
+      confirmSignatureStub.resolves({
+        ...tx,
+        r: new BN('1'),
+        s: new BN('2'),
+        v: new BN('3'),
+      } as any);
+
+      const account = await ledgerKeyring?.getDefaultAccount();
```

Comment:
Is the `?.` needed here?

Reply:
[6babec7](https://github.com/MetaMask/controllers/pull/862/commits/6babec76332e025630e0dbce6fc818a332a8e2e3)